flaky e2e controller-manager

https://github.com/volcano-sh/kthena/actions/runs/25804224060/job/75802326900?pr=1036
https://github.com/volcano-sh/kthena/actions/runs/25800731739/job/75789746454?pr=1035
https://github.com/FAUST-BENCHOU/kthena/actions/runs/25807322965/job/75813641668?pr=24
```
=== RUN   TestModelServingRollingUpdateMaxUnavailableWithBadImage
    model_serving_test.go:1025: Creating ModelServing with 6 replicas and maxUnavailable=2
    model_serving_test.go:1029: Waiting for initial ModelServing to be ready
    utils.go:41: Waiting for ModelServing to be ready...
    model_serving_test.go:1042: Updating ModelServing with bad image
    model_serving_test.go:1091: Monitoring rolling update with bad image for 60 seconds
panic: test timed out after 10m0s
	running tests:
		TestModelServingRollingUpdateMaxUnavailableWithBadImage (1m5s)

goroutine 420 [running]:
testing.(*M).startAlarm.func1()
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:2484 +0x394
created by time.goFunc
	/opt/hostedtoolcache/go/1.24.0/x64/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0002c1c00, {0x16ed10e?, 0xc000521a08?}, 0x1766fe0)
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:1859 +0x431
testing.runTests.func1(0xc0002c1c00)
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:2279 +0x37
testing.tRunner(0xc0002c1c00, 0xc000521b48)
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:1792 +0xf4
testing.runTests(0xc0001401b0, {0x21e39a0, 0x1e, 0x1e}, {0xc00044e2c0?, 0x2214d40?, 0x21f26c0?})
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:2277 +0x4b4
testing.(*M).Run(0xc00045e460)
	/opt/hostedtoolcache/go/1.24.0/x64/src/testing/testing.go:2142 +0x64a
github.com/volcano-sh/kthena/test/e2e/controller-manager.TestMain(0xc00045e460)
	/home/runner/work/kthena/kthena/test/e2e/controller-manager/test_suite_test.go:75 +0x2dc
main.main()
	_testmain.go:105 +0xa8
```

--- a/Makefile
+++ b/Makefile
@@ -105,7 +105,7 @@ test-e2e: ## Run all e2e tests sequentially (legacy).
 test-e2e-controller-manager: ## Run controller-manager e2e tests.
 	@command -v kind >/dev/null 2>&1 || { echo "Kind is not installed."; exit 1; }
 	@TEST_CATEGORY=controller-manager ./test/e2e/setup.sh
-	@KUBECONFIG=/tmp/kubeconfig-e2e go test -v -timeout=10m ./test/e2e/controller-manager/...
+	@KUBECONFIG=/tmp/kubeconfig-e2e go test -v -timeout=15m ./test/e2e/controller-manager/...
 
 .PHONY: test-e2e-router
 test-e2e-router: ## Run router e2e tests.

--- a/pkg/model-booster-controller/controller/model_booster_controller.go
+++ b/pkg/model-booster-controller/controller/model_booster_controller.go
@@ -33,6 +33,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	listerv1 "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
@@ -256,8 +257,24 @@ func (mc *ModelBoosterController) isModelServingActive(model *workload.ModelBoos
 
 // updateModelBoosterStatus updates model status.
 func (mc *ModelBoosterController) updateModelBoosterStatus(ctx context.Context, modelBooster *workload.ModelBooster) error {
-	modelBooster.Status.ObservedGeneration = modelBooster.Generation
-	if _, err := mc.client.WorkloadV1alpha1().ModelBoosters(modelBooster.Namespace).UpdateStatus(ctx, modelBooster, metav1.UpdateOptions{}); err != nil {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		latest, err := mc.modelBoosterLister.ModelBoosters(modelBooster.Namespace).Get(modelBooster.Name)
+		if err != nil {
+			return err
+		}
+		updated := latest.DeepCopy()
+		for i := range modelBooster.Status.Conditions {
+			meta.SetStatusCondition(&updated.Status.Conditions, modelBooster.Status.Conditions[i])
+		}
+		updated.Status.ObservedGeneration = updated.Generation
+		res, err := mc.client.WorkloadV1alpha1().ModelBoosters(updated.Namespace).UpdateStatus(ctx, updated, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		modelBooster.Status = res.Status
+		return nil
+	})
+	if err != nil {
 		klog.Errorf("update modelBooster status failed: %v", err)
 		return err
 	}

